tests/analysis/collation_token_stream_test.cpp

```diff
@@ -20,41 +20,317 @@
 /// @author Andrey Abramov
 ////////////////////////////////////////////////////////////////////////////////
 
-#include "tests_shared.hpp"
+#include "analysis/collation_token_stream.hpp"
 
 #include <unicode/coll.h>
 #include <unicode/sortkey.h>
+#include <unicode/locid.h>
 
-#include "analysis/collation_token_stream.hpp"
+#include "tests_shared.hpp"
 #include "utils/locale_utils.hpp"
+#include "velocypack/Parser.h"
+#include "velocypack/velocypack-aliases.h"
 
 TEST(collation_token_stream_test, consts) {
   static_assert("collation" == irs::type<irs::analysis::collation_token_stream>::name());
 }
 
 TEST(collation_token_stream_test, construct) {
+  // json
   {
     auto stream = irs::analysis::analyzers::get(
       "collation", irs::type<irs::text_format::json>::get(),
-      R"({ "locale": "en-EN.UTF-8" })");
+      R"({ "locale": {"language" : "en", "country": "US", "variant" : "UTF-8"}})");
     ASSERT_NE(nullptr, stream);
     ASSERT_EQ(irs::type<irs::analysis::collation_token_stream>::id(), stream->type());
   }
 
   {
     auto stream = irs::analysis::analyzers::get(
-      "collation", irs::type<irs::text_format::text>::get(),
-      "en-EN");
+      "collation", irs::type<irs::text_format::json>::get(),
+      R"({ "locale": {"language" : "en", "country": "US", "variant" : "phonebook"}})");
     ASSERT_NE(nullptr, stream);
     ASSERT_EQ(irs::type<irs::analysis::collation_token_stream>::id(), stream->type());
   }
 
+  {
+    auto stream = irs::analysis::analyzers::get(
+      "collation", irs::type<irs::text_format::json>::get(),
+      R"({ "locale": {"country": "US"}})");
+    ASSERT_EQ(nullptr, stream);
+  }
+
+  // vpack
+  {
+    std::string config = R"({ "locale": {"language" : "en", "country": "US", "variant" : "phonebook"}})";
+    auto in_vpack = VPackParser::fromJson(config.c_str(), config.size());
+    std::string in_str;
+    in_str.assign(in_vpack->slice().startAs<char>(), in_vpack->slice().byteSize());
+    std::string out_str;
+    auto stream = irs::analysis::analyzers::get("collation", irs::type<irs::text_format::vpack>::get(), in_str);
+    ASSERT_NE(nullptr, stream);
+    ASSERT_EQ(irs::type<irs::analysis::collation_token_stream>::id(), stream->type());
+  }
+
```

Comment:
Please add test cases:
language  and variant but no country
only variant
only country